protected.go

```diff
@@ -117,14 +179,19 @@ func (p *Protector) resolve(op ops.Op, network string, addr string) (*net.TCPAdd
 		log.Errorf("No IP address available: %v", err)
 		return nil, err
 	}
-	return &net.TCPAddr{IP: ipAddr, Port: port}, nil
+	return getProtectedAddr(network, ipAddr, port)
 }
 
 // Dial creates a new protected connection.
 // - syscall API calls are used to create and bind to the
 //   specified system device (this is primarily
 //   used for Android VpnService routing functionality)
-func (p *Protector) Dial(network, addr string, timeout time.Duration) (net.Conn, error) {
+func (p *Protector) Dial(network, addr string) (net.Conn, error) {
+	ctx, _ := context.WithTimeout(context.Background(), defaultDialTimeout)
```

Comment:
We need to call the returned cancel() function in a defer.

Reply:
Yeah, interesting.. that isn't a change I made (guess it's been that way awhile). Just updated it.